libp2p/protocols/kademlia/dhttypes.nim

```diff
@@ -0,0 +1,60 @@
+import chronicles
+import std/[tables, hashes]
+import results
+
+# TODO: make data formater for log-friendly form.
+
+type EntryKey* = object
+  data*: seq[byte]
+
+type EntryVal* = object
+  data*: seq[byte]
+
+type TimeStamp* = object
+  # Currently a string, because for some reason, that's what is chosen at the protobuf level
+  # TODO: convert between RFC3339 strings and use of integers (i.e. the _correct_ way)
+  ts*: string
+
+type EntryCandidate* = object
+  key*: EntryKey
+  val*: EntryVal
+
+type ValidatedEntry* = object
+  key*: EntryKey
+  val*: EntryVal
+
+type RecordVal* = object
+  val*: EntryVal
+  time*: TimeStamp
```

Comment:
All this is public for dev purposes. will need to make things private as appropriate

the separation of candidate vand valid entry is to do with the typestate pattern. the idea being is that the type being used encapsulates the invariants being upheld.

im still depating how to approach enforcing valid entries